test/Swaps.t.sol

```diff
@@ -5,289 +5,327 @@ import "forge-std/Test.sol";
 
-        assertEq(sDai.balanceOf(buyer),        80e18);
-        assertEq(sDai.balanceOf(address(psm)), 20e18);
+    function setUp() public override {
+        super.setUp();
 
-        assertEq(usdc.balanceOf(buyer),        0);
-        assertEq(usdc.balanceOf(address(psm)), 200e6);
+        dai.mint(address(psm),  DAI_TOKEN_MAX);
+        usdc.mint(address(psm), USDC_TOKEN_MAX);
+        sDai.mint(address(psm), SDAI_TOKEN_MAX);
     }
 
-    function test_swapAssetZeroToOne_differentReceiver() public assertAtomicPsmValueDoesNotChange {
-        usdc.mint(buyer, 100e6);
+    function _swapTest(
+        MockERC20 assetIn,
+        MockERC20 assetOut,
+        uint256 amountIn,
+        uint256 amountOut,
+        address swapper,
+        address receiver
+    ) internal {
+        // 1 trillion of each token corresponds to MAX values
+        uint256 psmAssetInBalance  = 1_000_000_000_000 * 10 ** assetIn.decimals();
+        uint256 psmAssetOutBalance = 1_000_000_000_000 * 10 ** assetOut.decimals();
 
-        vm.startPrank(buyer);
+        assetIn.mint(swapper, amountIn);
 
-        usdc.approve(address(psm), 100e6);
+        vm.startPrank(swapper);
+
+        assetIn.approve(address(psm), amountIn);
 
-        assertEq(usdc.allowance(buyer, address(psm)), 100e6);
+        assertEq(assetIn.allowance(swapper, address(psm)), amountIn);
 
-        assertEq(sDai.balanceOf(buyer),        0);
-        assertEq(sDai.balanceOf(receiver),     0);
-        assertEq(sDai.balanceOf(address(psm)), 100e18);
+        assertEq(assetIn.balanceOf(swapper),      amountIn);
+        assertEq(assetIn.balanceOf(address(psm)), psmAssetInBalance);
 
-        assertEq(usdc.balanceOf(buyer),        100e6);
-        assertEq(usdc.balanceOf(receiver),     0);
-        assertEq(usdc.balanceOf(address(psm)), 100e6);
+        assertEq(assetOut.balanceOf(receiver),     0);
+        assertEq(assetOut.balanceOf(address(psm)), psmAssetOutBalance);
 
-        psm.swapAssetZeroToOne(100e6, 80e18, receiver);
+        psm.swap(address(assetIn), address(assetOut), amountIn, amountOut, receiver);
 
-        assertEq(usdc.allowance(buyer, address(psm)), 0);
+        assertEq(assetIn.allowance(swapper, address(psm)), 0);
 
-        assertEq(sDai.balanceOf(buyer),        0);
-        assertEq(sDai.balanceOf(receiver),     80e18);
-        assertEq(sDai.balanceOf(address(psm)), 20e18);
+        assertEq(assetIn.balanceOf(swapper),      0);
+        assertEq(assetIn.balanceOf(address(psm)), psmAssetInBalance + amountIn);
 
-        assertEq(usdc.balanceOf(buyer),        0);
-        assertEq(usdc.balanceOf(receiver),     0);
-        assertEq(usdc.balanceOf(address(psm)), 200e6);
+        assertEq(assetOut.balanceOf(receiver),     amountOut);
+        assertEq(assetOut.balanceOf(address(psm)), psmAssetOutBalance - amountOut);
     }
 
 }
 
-contract PSMSwapAssetOneToZeroTests is PSMTestBase {
+contract PSMSwapDaiAssetInTests is PSMSuccessTestsBase {
```

Comment:
This contract is labelled as `PSMSwapDaiAssetInTests`, but also contains the USDC and sDAI tests. I'm assuming this is a mistake and you want `PSMSwapUSDCAssetInTests`, etc.